test/test.go

```diff
@@ -29,6 +30,47 @@ func AssertCanvasTappableAt(t *testing.T, c fyne.Canvas, pos fyne.Position) bool
 	return true
 }
 
+// AssertObjectRendersToImage asserts that the given `CanvasObject` renders the same image as the one stored in the master file.
+// The theme used is the standard test theme which may look different to how it shows on your device.
+// The master filename is relative to the `testdata` directory which is relative to the test.
+// The test `t` fails if the given image is not equal to the loaded master image.
+// In this case the given image is written into a file in `testdata/failed/<masterFilename>` (relative to the test).
+// This path is also reported, thus the file can be used as new master.
+//
+// Since 2.3
+func AssertObjectRendersToImage(t *testing.T, masterFilename string, o fyne.CanvasObject, msgAndArgs ...interface{}) bool {
+	c := NewCanvasWithPainter(software.NewPainter())
+	c.SetPadded(false)
+	size := o.MinSize().Max(o.Size())
+	c.SetContent(o)
+	c.Resize(size) // ensure we are large enough for current size
+
+	img := c.Capture()
```

Comment:
nit: I’d inline this or maybe even call `AssertRendersToImage` but that’s more a personal preference.

Reply:
Good point, I missed that due to the order I coded it in I think, thanks